src/angularjs/src/app/places/compare/compare.controller.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * @ngdoc controller
+ * @name pfb.analysis-jobs.detail.controller:CompareController
+ *
+ * @description
+ * Controller for showing details about an analysis job
+ *
+ */
+(function() {
+    'use strict';
+
+    /** @ngInject */
+    function CompareController($stateParams, Neighborhood, AnalysisJob, $log, $q, $state) {
+        var ctl = this;
+
+        initialize();
+
+        function initialize() {
+            ctl.places = new Array(3);
+            ctl.getPlace = getPlace;
+            ctl.clearSelection = clearSelection;
+
+            getPlaces([$stateParams.place1, $stateParams.place2, $stateParams.place3]);
+        }
+
+        function getPlaces(uuids) {
+            var promises = _.map(uuids, function(uuid, offset) {
+                return getPlace(offset, uuid);
+            });
+
+            // do not display any place until all places have been retrieved
+            $q.all(promises).then(function(results) {
+                ctl.places = results;
```

Comment:
It was mainly the individual cards loading in chunks that looked bad to me, so loading each place into `ctl.places[offset]` as it arrives would be fine as well, I think.  Not sure which is better, but the difference is minor and this is here, so this is good.